File: src/main/java/tool/dbcleanup/DBCleanupDialog.java
// %198737965:de.hattrickorganizer.gui.menu.option%
package tool.dbcleanup;

import core.gui.HOMainFrame;
import core.gui.comp.panel.ImagePanel;
import core.model.HOVerwaltung;

import java.awt.BorderLayout;
import java.awt.Dimension;
import java.awt.Font;
import java.awt.GridBagConstraints;
import java.awt.GridBagLayout;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

import javax.swing.JButton;
import javax.swing.JCheckBox;
import javax.swing.JDialog;
import javax.swing.JFrame;
import javax.swing.JLabel;
import javax.swing.JPanel;
import javax.swing.JTextArea;



/**
 * Database Cleanup Dialog
 *
 * @author flattermann <HO@flattermann.net>
 */
class DBCleanupDialog extends JDialog implements ActionListener {

	private static final long serialVersionUID = 3533368597781557223L;
	private JButton m_jbCleanupNow = new JButton(HOVerwaltung.instance().getLanguageString("dbcleanup.cleanupnow"));
	private JButton m_jbCancel = new JButton(HOVerwaltung.instance().getLanguageString("ls.button.cancel"));
	private DBCleanupTool cleanupTool;

	private JTextArea textIntro = new JTextArea (HOVerwaltung.instance().getLanguageString("dbcleanup.intro"));

	private JLabel labelOwnMatches = new JLabel (HOVerwaltung.instance().getLanguageString("dbcleanup.ownMatches"));
	private JLabel labelOwnFriendlies = new JLabel (HOVerwaltung.instance().getLanguageString("dbcleanup.ownFriendlies"));
	private JLabel labelOtherMatches = new JLabel (HOVerwaltung.instance().getLanguageString("dbcleanup.otherMatches"));
	private JLabel labelOtherFriendlies = new JLabel (HOVerwaltung.instance().getLanguageString("dbcleanup.otherFriendlies"));

	private WeekSelectionPanel wsp_OwnMatches =	new WeekSelectionPanel (DBCleanupTool.REMOVE_NONE);
	private WeekSelectionPanel wsp_OwnFriendlies = new WeekSelectionPanel (DBCleanupTool.REMOVE_NONE);
	private WeekSelectionPanel wsp_OtherMatches = new WeekSelectionPanel (16);
	private WeekSelectionPanel wsp_OtherFriendlies = new WeekSelectionPanel (8);

	private JLabel labelHrf = new JLabel (HOVerwaltung.instance().getLanguageString("dbcleanup.hrf"));
//	private WeekSelectionPanel wsp_Hrf = new WeekSelectionPanel (DBCleanupTool.REMOVE_NONE, false);

//	private JLabel labelHrfAutoremove = new JLabel (HOVerwaltung.instance().getLanguageString("dbcleanup.hrfAutoremove"));
	private JCheckBox m_jcbHrfAutoremove = new JCheckBox(HOVerwaltung.instance().getLanguageString("dbcleanup.hrfAutoremove"));

	//~ Constructors -------------------------------------------------------------------------------

	/**
	 * Creates a new DBCleanupDialog object.
	 */
	DBCleanupDialog(JFrame owner, DBCleanupTool cleanupTool) {
		super(owner,
				HOVerwaltung.instance().getLanguageString("ls.menu.file.database.databasecleanup"),
				true);
		this.cleanupTool = cleanupTool;
		setDefaultCloseOperation(JDialog.DISPOSE_ON_CLOSE);
		initComponents();
	}

	//~ Methods ------------------------------------------------------------------------------------

	private void initComponents() {
//		setContentPane(new de.hattrickorganizer.gui.templates.ImagePanel());
		getContentPane().setLayout(new BorderLayout());

		textIntro.setAlignmentX(CENTER_ALIGNMENT);
		textIntro.setEditable(false);
		textIntro.setWrapStyleWord(true);
		textIntro.setLineWrap(true);
		JPanel weekSelectionPanel = new JPanel(new GridBagLayout());

		labelOwnMatches.setFont(labelOwnMatches.getFont().deriveFont(Font.BOLD));
		labelOwnFriendlies.setFont(labelOwnFriendlies.getFont().deriveFont(Font.BOLD));
		labelOtherMatches.setFont(labelOtherMatches.getFont().deriveFont(Font.BOLD));
		labelOtherFriendlies.setFont(labelOtherFriendlies.getFont().deriveFont(Font.BOLD));
		labelHrf.setFont(labelHrf.getFont().deriveFont(Font.BOLD));
//		labelHrfAutoremove.setFont(labelHrfAutoremove.getFont().deriveFont(Font.BOLD));

		m_jcbHrfAutoremove.setSelected(true);

		GridBagConstraints c = new GridBagConstraints();

		c.gridx=0;
		c.gridy=0;
		c.anchor=GridBagConstraints.LINE_START;
		c.ipadx=20;

		weekSelectionPanel.add (labelOwnMatches, c);
		c.gridy++;
		weekSelectionPanel.add (labelOwnFriendlies, c);
		c.gridy++;
		weekSelectionPanel.add (labelOtherMatches, c);
		c.gridy++;
		weekSelectionPanel.add (labelOtherFriendlies, c);
		c.gridy++;
//		weekSelectionPanel.add (labelHrfAutoremove, c);
//		c.gridy++;
		weekSelectionPanel.add (labelHrf, c);

		c.gridx=1;
		c.gridy=0;
		weekSelectionPanel.add (wsp_OwnMatches, c);
		c.gridy++;
		weekSelectionPanel.add (wsp_OwnFriendlies, c);
		c.gridy++;
		weekSelectionPanel.add (wsp_OtherMatches, c);
		c.gridy++;
		weekSelectionPanel.add (wsp_OtherFriendlies, c);
		c.gridy++;
		weekSelectionPanel.add(m_jcbHrfAutoremove, c);
//		c.gridy++;
//		weekSelectionPanel.add (wsp_Hrf, c);


		ImagePanel m_jpButtonPanel = new ImagePanel();
		// Add Buttons
		m_jpButtonPanel.add(m_jbCleanupNow);
		m_jbCleanupNow.setFont(m_jbCleanupNow.getFont().deriveFont(Font.BOLD));
		m_jpButtonPanel.add(m_jbCancel);

		m_jbCleanupNow.addActionListener(this);
		m_jbCancel.addActionListener(this);

		getContentPane().add(textIntro, BorderLayout.NORTH);
		getContentPane().add(weekSelectionPanel, BorderLayout.CENTER);
		getContentPane().add(m_jpButtonPanel, BorderLayout.SOUTH);

		pack();

        final Dimension size = HOMainFrame.INSTANCE.getToolkit().getScreenSize();

        if (size.width > this.getSize().width) {
            //Mittig positionieren
            this.setLocation((size.width / 2) - (this.getSize().width / 2),
                             (size.height / 2) - (this.getSize().height / 2));
        }

		setVisible(true);
	}

	@Override
	public void actionPerformed(ActionEvent e) {
		if (e.getSource().equals(m_jbCleanupNow)) {
			cleanupTool.cleanupMatches (wsp_OwnMatches.getWeeks(), wsp_OwnFriendlies.getWeeks(), wsp_OtherMatches.getWeeks(), wsp_OtherFriendlies.getWeeks());
			cleanupTool.cleanupHRFs (DBCleanupTool.REMOVE_NONE, m_jcbHrfAutoremove.isSelected());
			setVisible(false);
		} else if (e.getSource().equals(m_jbCancel)) {
			setVisible(false);
		}
	}
}

File: src/main/java/tool/dbcleanup/DBCleanupDialog.kt
package tool.dbcleanup

import core.gui.HOMainFrame
import core.model.HOVerwaltung
import java.awt.*
import javax.swing.*

/**
 * Database Cleanup Dialog
 *
 * @author flattermann <HO@flattermann.net>
 */
internal class DBCleanupDialog(owner: JFrame?, private val cleanupTool: DBCleanupTool) : JDialog(
    owner,
    HOVerwaltung.instance().getLanguageString("ls.menu.file.database.databasecleanup"),
    true
) {
    private val mainPanelOwnMatches = WeekSelectionPanel(DBCleanupTool.REMOVE_NONE)
    private val mainPanelOtherMatches = WeekSelectionPanel(16)

    private val ownMatchesTypeSelectionPanel = MatchTypeSelectionPanel()
    private val othersMatchesTypeSelectionPanel = MatchTypeSelectionPanel()

    //~ Constructors -------------------------------------------------------------------------------
    /**
     * Creates a new DBCleanupDialog object.
     */
    init {
        defaultCloseOperation = DISPOSE_ON_CLOSE
        initComponents()
    }

    //~ Methods ------------------------------------------------------------------------------------
    private fun initComponents() {
        val mainPanel = JPanel()
        val gbc = GridBagConstraints()
        gbc.gridx = 0
        gbc.gridy = 0
        gbc.fill = GridBagConstraints.BOTH
        gbc.ipady = 5
        gbc.ipadx = 10

        mainPanel.layout = GridBagLayout()
        mainPanel.border = BorderFactory.createEmptyBorder(10, 10, 10, 10)

        gbc.gridwidth = 2
        val textIntro = JLabel(
            "<html>" + HOVerwaltung.instance().getLanguageString("dbcleanup.intro")
                .replace("\n", "<br>") + "</html>"
        )
        mainPanel.add(textIntro, gbc)

        gbc.gridwidth = 1
        val labelOwnMatches = JLabel(HOVerwaltung.instance().getLanguageString("dbcleanup.yourMatches"))
        labelOwnMatches.font = labelOwnMatches.font.deriveFont(Font.BOLD)

        val labelOtherMatches = JLabel(HOVerwaltung.instance().getLanguageString("dbcleanup.otherTeamsMatches"))
        labelOtherMatches.font = labelOtherMatches.font.deriveFont(Font.BOLD)

        val labelHrf = JLabel(HOVerwaltung.instance().getLanguageString("dbcleanup.hrf"))
        labelHrf.font = labelHrf.font.deriveFont(Font.BOLD)

        val hrfAutoRemove = JCheckBox(HOVerwaltung.instance().getLanguageString("dbcleanup.hrfSmartRemove"))
        hrfAutoRemove.isSelected = true

        gbc.gridx = 0
        gbc.gridy = 1
        gbc.weightx = 1.0
        gbc.fill = GridBagConstraints.BOTH
        gbc.anchor = GridBagConstraints.LINE_START

        mainPanel.add(labelOwnMatches, gbc)
        gbc.gridy = 3
        mainPanel.add(labelOtherMatches, gbc)
        gbc.gridy = 5
        mainPanel.add(labelHrf, gbc)

        gbc.gridx = 1
        gbc.gridy = 1
        mainPanel.add(mainPanelOwnMatches, gbc)
        gbc.gridy++
        mainPanel.add(ownMatchesTypeSelectionPanel, gbc)
        gbc.gridy++
        mainPanel.add(mainPanelOtherMatches, gbc)
        gbc.gridy++
        mainPanel.add(othersMatchesTypeSelectionPanel, gbc)
        gbc.gridy++
        mainPanel.add(hrfAutoRemove, gbc)

       addNewStrut(gbc, mainPanel)

        // Add current statistics on DB Records
        gbc.gridy++
        gbc.gridx = 0
        gbc.anchor = GridBagConstraints.WEST
        gbc.weightx = 1.0
        mainPanel.add(JLabel(HOVerwaltung.instance().getLanguageString("dbcleanup.numMatches")), gbc)

        gbc.gridx = 1
        gbc.anchor = GridBagConstraints.LINE_END
        gbc.weightx = 3.0
        mainPanel.add(JLabel(cleanupTool.getMatchesCount().toString()), gbc)

        gbc.gridx = 0
        gbc.gridy++
        gbc.anchor = GridBagConstraints.LINE_START
        mainPanel.add(JLabel(HOVerwaltung.instance().getLanguageString("dbcleanup.numHrfs")), gbc)
        gbc.gridx = 1
        mainPanel.add(JLabel(cleanupTool.getHrfCount().toString()), gbc)
        gbc.gridy++

        addNewStrut(gbc, mainPanel)

        gbc.gridwidth = 2

        // Add Buttons
        val buttonPanel = JPanel()
        val cleanupNowButton = JButton(HOVerwaltung.instance().getLanguageString("dbcleanup.cleanupnow"))
        cleanupNowButton.font = cleanupNowButton.font.deriveFont(Font.BOLD)
        cleanupNowButton.addActionListener {
            cleanupTool.cleanupMatches(
                CleanupDetails(
                    ownMatchesTypeSelectionPanel.getSelectedMatchTypes(),
                    othersMatchesTypeSelectionPanel.getSelectedMatchTypes(),
                    mainPanelOwnMatches.getWeeks(),
                    mainPanelOtherMatches.getWeeks())
            )
            cleanupTool.cleanupHRFs(DBCleanupTool.REMOVE_NONE, hrfAutoRemove.isSelected)
            isVisible = false
        }

        val cancelButton = JButton(HOVerwaltung.instance().getLanguageString("ls.button.cancel"))
        cancelButton.addActionListener {
            isVisible = false
        }
        buttonPanel.add(cleanupNowButton)
        buttonPanel.add(cancelButton)

        mainPanel.add(buttonPanel, gbc)

        contentPane.layout = BorderLayout()
        contentPane.add(mainPanel, BorderLayout.CENTER)

        pack()

        val screenSize = HOMainFrame.instance().toolkit.screenSize
        if (screenSize.width > size.width) {
            // Place in the middle
            this.setLocation(
                (screenSize.width / 2) - (size.width / 2),
                (screenSize.height / 2) - (size.height / 2)
            )
        }

        isVisible = true
    }

    private fun addNewStrut(gbc: GridBagConstraints, mainPanel: JPanel, height: Int = 20) {
        gbc.gridy++
        gbc.gridwidth = 2
        gbc.gridx = 0

        val strut = JPanel()
        strut.preferredSize = Dimension(20, height)
        mainPanel.add(strut, gbc)

        gbc.gridwidth = 1
        gbc.gridy++
    }
}

